hydroflow/examples/two_pc/coordinator.rs

```diff
@@ -118,38 +134,72 @@ pub(crate) async fn run_coordinator(opts: Opts, subordinates: Vec<String>) {
             .push_to(p1_send_push),
     );
 
+    // Count the size of subordinates.
+    hf.add_subgraph_stratified(
+        "count the size of subordinates",
+        0,
+        subordinates_out_tee_4_pull
+            .flatten()
+            .map_scan(0u32, |counter, _| {
+                *counter += 1;
+                *counter
+            })
+            .pull_to_push()
+            .map(Some)
+            .push_to(subordinate_count_1_push),
+    );
+
+    // Take the max of subordinate counts.
+    hf.add_subgraph_stratified(
+        "max subordinate counts",
+        1,
+        subordinate_count_1_pull
+            .filter_map(|batch| batch.into_iter().max())
+            .pull_to_push()
+            .map(Some)
+            .tee(
+                subordinate_max_count_tee_1_push,
+                subordinate_max_count_tee_2_push,
+            ),
+    );
+
     // Phase 1 Response and Phase 2 Request:
     // Receive and count subordinate responses until we have them all.
     // Once we have them all, assess the unanimous vote condition for commit and
     // send either a Commit or Abort message.
     // HACK: We move the vote_ctr((xid, mid), count) HashMap into a map operator as flow state.
     // Should be done with a groupby.
-    let mut vote_ctr = HashMap::new();
-    hf.add_subgraph(
+    hf.add_subgraph_stratified(
         "collect votes and send command",
+        1,
         p1_recv_pull
             .flatten()
             .map(move |msg| {
                 println!("{:?}", msg);
                 msg
             })
-            .filter_map(move |msg| {
-                *vote_ctr.entry((msg.xid, msg.mtype)).or_insert(0) += 1;
-                // HACK: the constant 3 here is the length of the subordinates list!
-                // Should be computed on bootstrap tick and joined in here.
-                if vote_ctr.get(&(msg.xid, MsgType::Commit)).unwrap_or(&0)
-                    + vote_ctr.get(&(msg.xid, MsgType::Abort)).unwrap_or(&0)
-                    == 3
-                {
+            .map_scan(
+                HashMap::<u16, (u32, SubordResponse, bool)>::new(),
+                |txs, msg| {
```

Comment:
add comment and/or use more meaning  variable names so we understand the HashMap is holding ballot-counts for all transactions, partitioned by transactionId.